pkg/clusteragent/externalmetrics/provider.go

```diff
@@ -136,14 +135,14 @@ func (p *datadogMetricProvider) getExternalMetric(namespace string, metricSelect
 		parsed = true
 	}
 	if !parsed {
-		return nil, fmt.Errorf("ExternalMetric does not follow DatadogMetric format")
+		return nil, log.Errorf("ExternalMetric does not follow DatadogMetric format: %s", info.Metric)
 	}
 
 	datadogMetric := p.store.Get(datadogMetricID)
-	log.Debugf("DatadogMetric from store: %v", datadogMetric)
+	log.Tracef("DatadogMetric from store: %v", datadogMetric)
 
 	if datadogMetric == nil {
-		return nil, fmt.Errorf("DatadogMetric not found for metric name: %s, datadogmetricid: %s", info.Metric, datadogMetricID)
+		return nil, log.Errorf("DatadogMetric not found for metric name: %s, datadogmetricid: %s", info.Metric, datadogMetricID)
```

Comment:
could you add a comment to say that this should not happen. like you did in another error case

Reply:
It can happen if a user puts a DatadogMetric reference that does not exist in an HPA. Could be a warning as well.

Reply:
Warning seems reasonable